src/canmatrix/canmatrix.py

```diff
@@ -551,6 +552,48 @@ def pack_bitstring(length, is_float, value, signed):
 
     return bitstring
 
+@attr.s(cmp=False)
+class Arbitration_Id(object):
+    standard_id_mask = ((1 << 11) - 1)
+    extended_id_mask = ((1 << 29) - 1)
+    compound_extended_mask = (1 << 31)
+
+    id = attr.ib(default=None)
+    extended = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        if self.extended is None or self.extended:
+            mask = self.extended_id_mask
+        else:
+            mask = self.standard_id_mask
+
+        if self.id != self.id & mask:
+            raise ArbitrationIdOutOfRange('ID out of range')
+
+    @classmethod
+    def from_compound_integer(cls, i):
```

Comment:
```suggestion
    def from_compound_integer(cls, i, extended=True):
```

And associated changes inside the function.

Reply:
... the `compound_integer` already has the `extended`information in bit `31` - so no need for the extended attribute. Or do I miss something?

Reply:
:[ oh right, it's _compound_...  I'm really not on my game today, sorry.